cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/SparkBatchSinkContext.java

```diff
@@ -32,11 +32,14 @@
 public class SparkBatchSinkContext extends AbstractSparkBatchContext implements BatchSinkContext {
 
   private final SparkBatchSinkFactory sinkFactory;
+  private final String aliasSuffix;
+  private int outputCount = 0;
 
   public SparkBatchSinkContext(SparkBatchSinkFactory sinkFactory, SparkClientContext sparkContext,
-                               LookupProvider lookupProvider, String stageId) {
+                               LookupProvider lookupProvider, String stageId, String aliasSuffix) {
```

Comment:
The outputCount is within a particular sink. A given sink can have multiple outputs. Hence outputCount is required. But we also need a suffix that makes sure we have unique suffix across all batch sinks in a MapReduce/Spark job.